src/sentry/event_manager.py

```diff
@@ -1620,11 +1622,37 @@ def _save_aggregate(
         # _save_aggregate had races around group creation which made this race
         # more user visible. For more context, see 84c6f75a and d0e22787, as
         # well as GH-5085.
-        GroupHash.objects.filter(id__in=[h.id for h in new_hashes]).exclude(
-            state=GroupHash.State.LOCKED_IN_MIGRATION
-        ).update(group=group)
-
-    is_regression = _process_existing_aggregate(
+        with transaction.atomic(using="default"):
```

Comment:
Let's not hardcode the DB connection string and instead use db router.
```
using=router.db_for_write(GroupHash)
```
e.g. https://github.com/getsentry/sentry/blob/master/src/sentry/models/releasefile.py#L290-L295